src/generator/mod.rs

```diff
@@ -0,0 +1,204 @@
+pub mod default;
+pub mod utils;
+
+use crate::generator::utils::{default_format_comment, default_format_name, default_format_type};
+use crate::parser::types::{
+    Alias, Enum, EnumCase, Import, RsEntity, Struct, StructField, TupleStruct, TypeModifier,
+};
+use roxmltree::Namespace;
+use std::borrow::Cow;
+
+pub trait Generator<'input> {
+    fn target_ns(&self) -> &Option<Namespace<'_>>;
+
+    fn tuple_struct_macro(&self, _: &TupleStruct) -> Cow<'static, str> {
+        "".into()
+    }
+    fn struct_macro(&self, _: &Struct) -> Cow<'static, str> {
+        "".into()
+    }
+    fn enum_macro(&self, _: &Enum) -> Cow<'static, str> {
+        "".into()
+    }
+    fn alias_macro(&self, _: &Alias) -> Cow<'static, str> {
+        "".into()
+    }
+    fn struct_field_macro(&self, _: &StructField) -> Cow<'static, str> {
+        "".into()
+    }
+    fn enum_case_macro(&self, _: &EnumCase) -> Cow<'static, str> {
+        "".into()
+    }
+
+    fn get_rs_entity(&self, entity: &RsEntity) -> String {
```

Comment:
This name is confusing. I suggest renaming it to "gen_rs_entity", or "get_code_for_rs_entity", or something like that. The same for the bunch of following functions in this file.